geps/gep-2198.md

```diff
@@ -0,0 +1,105 @@
+  
+- Kong:
+
+  - [https://docs.konghq.com/hub/kong-inc/jwt/configuration/](https://docs.konghq.com/hub/kong-inc/jwt/configuration/)
+
+- Envoy Gateway:
+
+  - [https://gateway.envoyproxy.io/latest/api/extension_types.html#jwtauthenticationfilterprovider](https://gateway.envoyproxy.io/latest/api/extension_types.html#jwtauthenticationfilterprovider)
+
+- Istio:
+
+  - [https://istio.io/latest/docs/tasks/security/authorization/authz-jwt/](https://istio.io/latest/docs/tasks/security/authorization/authz-jwt/)
+  - [https://istio.io/latest/docs/tasks/security/authentication/jwt-route/](https://istio.io/latest/docs/tasks/security/authentication/jwt-route/)
+  - [https://istio.io/latest/docs/tasks/security/authentication/claim-to-header/](https://istio.io/latest/docs/tasks/security/authentication/claim-to-header/)
+
+## Properties of a JWT Validation Policy
+
+The validation policy is two fold, it establishes the relationship with the Token and how to validate it.
+According to [https://datatracker.ietf.org/doc/html/rfc7519#section-4.1](https://datatracker.ietf.org/doc/html/rfc7519#section-4.1) claims cannot be assumed mandatory.  At a minimum a claim should be able to be validated following the process outlined here which only requires a method to decrypt if necessary: [https://datatracker.ietf.org/doc/html/rfc7519#section-7.2](https://datatracker.ietf.org/doc/html/rfc7519#section-7.2)
+
+The statements made above point to the primary points of divergence between the JWT validation implementations in the prior art section.
+
+Some very high level proposed properties to open the conversation based on comparing the implementations and reviewing the RFC.
+
+- name / realm (required) - the name / realm of the JWT, this is commonly a string representing a specific application.  It is seen used in response and log messages.
+- secret (optional) - The name of the Kubernetes secret that stores the JWK.
+- jwksURI (optional) - The remote URI where the request will be sent to retrieve JSON Web Key set
+- keyCache (extension) - Enables in-memory caching of JWKS (JSON Web Key Sets) that are obtained from the jwksURI and sets a valid time for expiration.
+- token (extension) - The token specifies a variable that contains the JSON Web Token in the request. By default the JWT is passed in the Authorization header as a Bearer Token. JWT may be also passed as a cookie or a part of a query string, for example: $cookie_auth_token. Accepted variables are $http_, $arg_, $cookie_.
+
+Example patterns:
+name
+secret
+
+name
+jwksURI
+
+name
+token
+jwksURI
+keycache
+
+## Properties of a JWT Claims (authorization) Policy
+
+- token - the token that needs to be evaluated (this is the result of the validation Policy or an OIDC workflow)
+- claimsMatch - an array of claim properties (exists, not exists) or match value that must match to allow the request to proceed.
```

Comment:
I think claim match is already in the realm of authorization. 

It doesn't scale and doesn't work well in practice to have 'name=user1' ... or name=user100' in policies that specify which issuers to trust. 

And there are many other authn methods ( access tokens with Oauth, client certs, user:password, apikeys ) that result in 'claims' associated with a principal. We shouldn't have a duplicated or divergent authorization depending on how a user may choose or be able to authenticate.